pytensor/link/jax/dispatch/random.py

```diff
@@ -394,21 +395,44 @@ def sample_fn(rng_key, size, dtype, n, p):
 
 @jax_sample_fn.register(ptr.MultinomialRV)
 def jax_sample_fn_multinomial(op, node):
-    if not numpyro_available:
-        raise NotImplementedError(
-            f"No JAX implementation for the given distribution: {op.name}. "
-            "Implementation is available if NumPyro is installed."
-        )
-
-    from numpyro.distributions.util import multinomial
-
     def sample_fn(rng_key, size, dtype, n, p):
-        sample = multinomial(key=rng_key, n=n, p=p, shape=size)
+        sample = _jax_multinomial(key=rng_key, n=n, p=p, size=size)
         return sample
 
     return sample_fn
 
 
+def _jax_multinomial(n, p, size=None, key=None):
```

Comment:
You can inline this in the dispatch for MultinomialRV. Also key=None is not really valid but we don't need default parameters anyway